Mathlib/Analysis/SpecialFunctions/Log/Deriv.lean

```diff
@@ -319,4 +319,25 @@ theorem hasSum_log_one_add_inv {a : ℝ} (h : 0 < a) :
     linarith
   · field_simp
 
+/-- Expansion of `log (1 + a)` as a series in powers of `a / (a + 2)`. -/
+theorem hasSum_log_one_add {a : ℝ} (h : 0 ≤ a) :
+    HasSum (fun k : ℕ => (2 : ℝ) * (1 / (2 * k + 1)) * (a / (a + 2)) ^ (2 * k + 1))
+      (log (1 + a)) := by
+  by_cases ha0 : a = 0
+  · simp [ha0, hasSum_zero]
+  · have key := hasSum_log_one_add_inv (inv_pos.mpr (lt_of_le_of_ne h (Ne.symm ha0)))
+    rw [inv_inv, add_comm] at key
+    exact key.congr_fun fun k ↦ by field_simp
```

Comment:
There's nothing fundamentally wrong with these proofs, I just wanted to showcase a few tactic features in case you weren't familiar with them:

1. you can use `rfl` in `obtain` to substitute equalities automatically.
2. you can use `convert` to provide a term which almost matches the goal, and you will be left with differences. You can use `using n` to limit the depth of the match to `n` (so it isn't overly eager).

```suggestion
  obtain (rfl | ha0) := eq_or_ne a 0
  · simp [hasSum_zero]
  · convert hasSum_log_one_add_inv (inv_pos.mpr (lt_of_le_of_ne h ha0.symm)) using 4
    all_goals field_simp [add_comm]
```

Reply:
Thanks!